### PR TITLE
make exam service clean architecture #120

### PR DESCRIPTION
**Pull request by ryan01234keroro56789**
# Description

See the linked issues for the details.

- Make the exam services clean architecture
- Maybe reformat some files

### CODE REVIEW

File: Exam-Service/src/main/java/tw/waterball/judgegirl/examservice/repositories/ExamParticipationRepository.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package tw.waterball.judgegirl.examservice.repositories;
+
+import tw.waterball.judgegirl.entities.ExamParticipation;
+
+import javax.inject.Named;
+import java.util.List;
+
+@Named
```
Comment (Johnny850807) [inline Exam-Service/src/main/java/tw/waterball/judgegirl/examservice/repositories/ExamParticipationRepository.java:8]: This is unnecessary

File: Exam-Service/src/main/java/tw/waterball/judgegirl/examservice/usecases/CreateExamUseCase.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Data
+    @NoArgsConstructor
+    public static class Request {
+        @NotBlank
+        public String name;
+        @NotNull
+        public Date startTime;
+        @NotNull
+        public Date endTime;
```
Comment (Johnny850807) [inline Exam-Service/src/main/java/tw/waterball/judgegirl/examservice/usecases/CreateExamUseCase.java:39]: The validation annotations are unnecessary

File: Exam-Service/src/main/java/tw/waterball/judgegirl/examservice/usecases/CreateExamUseCase.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        exam.validate();
+        presenter.setExam(examRepository.save(exam));
+    }
+
+    public interface Presenter {
+        void setExam(Exam exam);
+    }
+
+    @Data
+    @NoArgsConstructor
```
Comment (Johnny850807) [inline Exam-Service/src/main/java/tw/waterball/judgegirl/examservice/usecases/CreateExamUseCase.java:32]: this is unnecessary

File: Exam-Service/src/main/java/tw/waterball/judgegirl/examservice/usecases/GetUpcomingExamUseCase.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+import javax.inject.Named;
+import javax.validation.constraints.NotNull;
+import java.util.Date;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Named
+public class GetUpcomingExamUseCase {
```
Comment (Johnny850807) [inline Exam-Service/src/main/java/tw/waterball/judgegirl/examservice/usecases/GetUpcomingExamUseCase.java:17]: `GetUpcomingExamsUseCase`

File: Exam-Service/src/main/java/tw/waterball/judgegirl/examservice/usecases/GetUpcomingExamUseCase.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        return examRepository.findByIdIn(examIds);
+    }
+
+    public interface Presenter {
+        void setExams(List<Exam> exams);
+    }
+
+    @Value
+    public static class Request {
+        @NotNull
```
Comment (Johnny850807) [inline Exam-Service/src/main/java/tw/waterball/judgegirl/examservice/usecases/GetUpcomingExamUseCase.java:46]: ditto

File: Spring-Boot/Spring-Boot-Exam/src/main/resources/application.yml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+server:
+  port: 80
+judge-girl:
+  student-service:
+    legacy-judge-girl:
+      base-url: https://judgegirl.csie.org
+      api-key: D9wBvKGSgC
+
```
Comment (Johnny850807) [inline Spring-Boot/Spring-Boot-Exam/src/main/resources/application.yml:8]: delete this file

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done!

**Review by Chaoyuuu — APPROVED** [review]
LGTM 👍🏻

File: .gitignore [hunk 7]
```diff
@@ -33,6 +33,8 @@ build/
 
 ### Secret ###
 infra.yml
+Spring-Boot/Spring-Boot-Exam/src/main/resources/application.yml
```
Comment (Johnny850807) [inline .gitignore:36]: you should track this file for sure.

and the file's content should be 
```yml
server:
  port: 80
```

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

**Review by Johnny850807 — APPROVED** [review]
LGTM

**Pull request merged**